velox/exec/tests/PartitionAndSerialize.h

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "velox/core/PlanNode.h"
+#include "velox/exec/Operator.h"
+
+namespace facebook::velox::exec {
+
+class PartitionAndSerializeNode : public core::PlanNode {
+ public:
+  PartitionAndSerializeNode(
+      const core::PlanNodeId& id,
+      std::vector<core::TypedExprPtr> keys,
+      int numPartitions,
+      RowTypePtr outputType,
+      core::PlanNodePtr source)
+      : core::PlanNode(id),
+        keys_{std::move(keys)},
+        numPartitions_{numPartitions},
+        outputType_{std::move(outputType)},
```

Comment:
Why does it need the outputType to be set from outside when it has to be ROW(p, d)

Reply:
The shape of the output type is fixed: INTEGER + VARBINARY, but the names of the individual fields are not. The output type allows the planner to choose symbol names in whatever way makes sense.